modules/module_utils/ansible_nailgun_cement.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+# (c) Bernhard Hopfenmüller 2017
+# (c) Matthias Dellweg 2017
+
+import re
+import yaml
+
+from nailgun.config import ServerConfig
+import nailgun.entities
+
+
+# Mix compare functionality into some entities as needed
+class EntityCompareMixin:
+
+    def __eq__(self, other):
+        return self.id == other.id
+
+    def __ne__(self, other):
+        return self.id != other.id
+
+
+class TemplateKind(EntityCompareMixin, nailgun.entities.TemplateKind):
+    pass
+
+
+class Organization(EntityCompareMixin, nailgun.entities.Organization):
+    pass
+
+
+class Location(EntityCompareMixin, nailgun.entities.Location):
+    pass
+
+
+# Connection helper
+def create_server(server_url, auth, verify_ssl):
+    ServerConfig(
+        url=server_url,
+        auth=auth,
+        verify=verify_ssl,
+    ).save()
+
+
+# Common functionality to manipulate entities
+def find_entity(entity_class, **kwargs):
+    return entity_class().search(
```

Comment:
I would love to make that more flexible! However, sadly nailgun is not flexible or consistent at all. We are currently discussing this in an issue [report](https://github.com/SatelliteQE/nailgun/issues/423).
Feel free to participate, if you have any ideas :)